### ISSUE TITLE
scatter(nan_color...) works for CairoMakie but not GLMakie

### ISSUE THREAD

**Issue by bjarthur** [MakieOrg/Makie.jl#3067]
the point at x=y=6 should be red in both plots below, but is not in the first one, which uses GLMakie:

```
julia> using GLMakie

julia> fig, ax, li = scatter(1:10, 1:10, color=[1:5; NaN; 7:10], nan_color=colorant"red")
```

<img width="512" alt="Screenshot 2023-07-14 at 4 37 50 PM" src="https://github.com/MakieOrg/Makie.jl/assets/1538268/92684f8d-df8b-4e17-8b8a-889805006608">

same exact code but with CairoMakie instead of GLMakie:

```
julia> using CairoMakie

julia> fig, ax, li = scatter(1:10, 1:10, color=[1:5; NaN; 7:10], nan_color=colorant"red")
```

<img width="912" alt="Screenshot 2023-07-14 at 4 38 49 PM" src="https://github.com/MakieOrg/Makie.jl/assets/1538268/5321834e-02dc-4ae7-8ea7-b13b59d4cab1">

```
(jl_lalGOb) pkg> st
Status `/private/var/folders/s5/8d629n5d7nsf37f60_91wzr40000gq/T/jl_lalGOb/Project.toml`
  [13f3f980] CairoMakie v0.10.6
  [e9467ef8] GLMakie v0.8.6

julia> VERSION
v"1.9.2"

julia> Sys.MACHINE
"arm64-apple-darwin22.4.0"
```

**Comment by bjarthur** [comment]
not shown above is that WGLMakie works fine.

**Comment by lazarusA** [comment]
I can reproduce this bug, :(.

**Comment by SimonDanisch** [comment]
Ah jeez, this must be a gpu specific issue ... nans are super Ill defined on the gpu, and we tried hard to implement them in a way to work around most driver issues, but seems like there are still problems

**Comment by SimonDanisch** [comment]
which gpu is this?

**Comment by SimonDanisch** [comment]
Ah m2 I suppose -.-

**Comment by bjarthur** [comment]
yes, M2.  sorry for not specifying that earlier.  didn't think it would make a difference.

thanks for looking into this!  let me know if i can help fix it.

**Comment by SimonDanisch** [comment]
Oh... I just noticed, that this wasn't actually implemented in GLMakie, even though everything was in place for it :-O 
Should be fixed in https://github.com/MakieOrg/Makie.jl/pull/2900, which should get merged soon!

**Comment by SimonDanisch** [comment]
Fixed in #2900
